include/aikido/planner/dart/ConfigurationToTSR.hpp

```diff
@@ -15,21 +15,43 @@ namespace dart {
 class ConfigurationToTSR : public Problem
 {
 public:
-  /// Constructor.
+  /// Constructor. Note that this constructor takes the start state from the
+  /// current state of the passed MetaSkeleton.
   ///
   /// \param[in] stateSpace State space.
+  /// \param[in] metaSkeleton MetaSkeleton that getStartState will return the
+  /// current state of when called.
   /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
   /// TSR.
   /// \param[in] maxSamples Maximum number of TSR samples to plan to.
-  /// \param[in] startState Start state.
   /// \param[in] goalTSR Goal TSR.
   /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
   /// \throw If \c stateSpace is not compatible with \c constraint's state
   /// space.
   ConfigurationToTSR(
       statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
       ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+      std::size_t maxSamples,
+      constraint::dart::ConstTSRPtr goalTSR,
+      constraint::ConstTestablePtr constraint);
+
+  /// Constructor. Note that this constructor sets the start state on
+  /// construction.
+  ///
+  /// \param[in] stateSpace State space.
+  /// \param[in] startState Start state to plan from.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// TSR.
+  /// \param[in] maxSamples Maximum number of TSR samples to plan to.
+  /// \param[in] goalTSR Goal TSR.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw If \c stateSpace is not compatible with \c constraint's state
+  /// space.
+  ConfigurationToTSR(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
       const statespace::dart::MetaSkeletonStateSpace::State* startState,
+      ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
       std::size_t maxSamples,
       constraint::dart::ConstTSRPtr goalTSR,
       constraint::ConstTestablePtr constraint);
```

Comment:
Nit: Let's pass `nullptr` to constraint as the default value for the same reason.